pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStage.java

```diff
@@ -102,31 +116,37 @@ public String toGroovy() {
         if (environment != null) {
             result.append(environment.toGroovy());
         }
-        result.append("steps {\n");
-        if (branches.size() > 1) {
-            result.append("parallel(");
-            boolean first = true;
-            for (ModelASTBranch branch: branches) {
-                if (first) {
-                    first = false;
-                } else {
-                    result.append(',');
+        if (branches.isEmpty() && parallelStages != null) {
+            result.append("parallelStages {\n");
```

Comment:
Compromise, basically. I believe that `parallel` is too overloaded for sure, the synonyms for `parallel` are completely new terms for Pipeline and so I'd prefer to avoid them (both to prevent the need for learning new terminology and, more importantly, to keep those terms free for possible later use!), and just `stages` is too overloaded as well. 

With this, `stages` ends up meaning "execute these stages linearly" while `parallel` means, as in Scripted, "execute these arbitrary blocks in parallel", and the new `parallelStages` means, fairly explicitly, "execute these stages concurrently". While I don't have any further use for those distinctions now, I can envision some possible use cases in the future.